sidekick/tests/test_utils.py

```diff
@@ -1,10 +1,92 @@
+import json
+
+import responses
 from django.test import TestCase
 from django.utils import timezone
+from mock import patch
 
 from sidekick import utils
+from sidekick.models import Organization
 
 
 class UtilsTests(TestCase):
+    def create_org(self):
+        return Organization.objects.create(
+            name="Test Organization",
+            url="http://localhost:8002/",
+            token="REPLACEME",
+            engage_url="http://whatsapp/",
+            engage_token="test-token",
+        )
+
+    def mock_rapidpro_contact_get(self, msisdn, count=1, wa_id=None):
+        urns = ["tel:{}".format(msisdn)]
+        if wa_id:
+            urns.append("whatsapp:{}".format(wa_id))
+
+        responses.add(
+            method=responses.GET,
+            url="http://localhost:8002/api/v2/contacts.json?urn=tel:{}".format(
+                msisdn
+            ),
+            json={
+                "count": count,
+                "next": None,
+                "previous": None,
+                "results": [
+                    {
+                        "uuid": "bfff9984-38f4-4e59-998d-3663ec3c650d",
+                        "name": "John Smith",
+                        "language": None,
+                        "group_uuids": ["04a4752b-0f49-480e-ae60-3a3f2bea485c"],
+                        "urns": urns,
+                        "fields": {"nickname": "Hannibal"},
+                        "blocked": False,
+                        "failed": False,
+                        "stopped": False,
+                        "modified_on": "2014-10-01T06:54:09.817Z",
+                        "created_on": "2014-10-01T06:54:09.817Z",
+                        "phone": "+27820001001",
+                        "groups": [],
+                    }
+                ]
+                * count,
+            },
+            status=200,
+        )
+
+    def mock_rapidpro_contact_post(self, msisdn, uuid=None, wa_id=None):
+        urns = ["tel:{}".format(msisdn)]
+        if wa_id:
+            urns.append("whatsapp:{}".format(wa_id))
+
+        url = "http://localhost:8002/api/v2/contacts.json"
+        if uuid:
+            url = "{}?uuid={}".format(url, uuid)
+
+        responses.add(
+            method=responses.POST,
+            url=url,
+            json={
+                "uuid": "09d23a05-47fe-11e4-bfe9-b8f6b119e9ab",
+                "name": "Ben Haggerty",
+                "language": "eng",
+                "urns": urns,
+                "groups": [
+                    {
+                        "name": "Devs",
+                        "uuid": "6685e933-26e1-4363-a468-8f7268ab63a9",
+                    }
+                ],
+                "fields": {"nickname": "Macklemore", "side_kick": "Ryan Lewis"},
```

Comment:
He he. Nice 😄

Reply:
Copied these tests from the Rapidpro python client. hehe